Incorrectly handled description in paths

Hi,

In this example
```
openapi: "3.0.3"
paths:
  /:
    description: Description for all methods under path <------ This causes problem
    get:
      responses:
        '200':
         description: root response
```

if I try to use this schema and run tests it fails with error
```
/lib/python3.7/site-packages/schemathesis/schemas.py:212: in get_all_endpoints
    or should_skip_by_tag(definition.get("tags"), self.tag)
E   AttributeError: 'str' object has no attribute 'get'
```

It's because schemathesis tries to read data for each node under path (`/:` in this example) and fails because description is string and not object

Not everything under `path` is method ([Specification](https://swagger.io/docs/specification/paths-and-operations/)), there can be also description and summary. 
Only specified methods should be used (`get, post, put, patch, delete, head, options, and trace`)

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -6,6 +6,11 @@ Changelog
 `Unreleased`_
 -------------
 
+Fixed
+~~~~~
+
+- Handling of fields in ``paths`` that are not operations, but allowed by the Open API spec. `#457`_
+
 `1.0.1`_ - 2020-04-01
 ---------------------
 
@@ -819,6 +824,7 @@ Fixed
 .. _0.3.0: https://github.com/kiwicom/schemathesis/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/kiwicom/schemathesis/compare/v0.1.0...v0.2.0
 
+.. _#457: https://github.com/kiwicom/schemathesis/issues/457
 .. _#450: https://github.com/kiwicom/schemathesis/issues/450
 .. _#448: https://github.com/kiwicom/schemathesis/issues/448
 .. _#440: https://github.com/kiwicom/schemathesis/issues/440

--- a/src/schemathesis/schemas.py
+++ b/src/schemathesis/schemas.py
@@ -169,6 +169,7 @@ class BaseSchema(Mapping):
 
 class SwaggerV20(BaseSchema):
     nullable_name = "x-nullable"
+    operations: Tuple[str, ...] = ("get", "put", "post", "delete", "options", "head", "patch")
 
     def __repr__(self) -> str:
         info = self.raw_schema["info"]
@@ -206,8 +207,7 @@ class SwaggerV20(BaseSchema):
                 for method, definition in methods.items():
                     # Only method definitions are parsed
                     if (
-                        method == "parameters"
-                        or method.startswith("x-")
+                        method not in self.operations
                         or should_skip_method(method, self.method)
                         or should_skip_by_tag(definition.get("tags"), self.tag)
                     ):
@@ -321,6 +321,7 @@ class SwaggerV20(BaseSchema):
 
 class OpenApi30(SwaggerV20):  # pylint: disable=too-many-ancestors
     nullable_name = "nullable"
+    operations = SwaggerV20.operations + ("trace",)
 
     @property
     def spec_version(self) -> str:

--- a/test/test_parametrization.py
+++ b/test/test_parametrization.py
@@ -418,8 +418,15 @@ def test_b(request, case):
     result.stdout.re_match_lines([".* 1 deselected / 2 selected", r".*\[POST:/v1/pets\]", r"Hypothesis calls: 2"])
 
 
-def test_custom_properties(testdir):
-    # When custom properties are present in endpoint definitions
+@pytest.mark.parametrize(
+    "schema_name, endpoint",
+    (
+        ("simple_swagger.yaml", {"/users": {"x-handler": "foo"}}),
+        ("simple_openapi.yaml", {"/users": {"x-handler": "foo", "description": "Text"}}),
+    ),
+)
+def test_custom_properties(testdir, schema_name, endpoint):
+    # When custom properties are present in endpoint definitions (e.g. vendor extensions or some other allowed fields)
     testdir.make_test(
         """
 @schema.parametrize()
@@ -427,7 +434,8 @@ def test_custom_properties(testdir):
 def test_(request, case):
     request.config.HYPOTHESIS_CASES += 1
     """,
-        paths={"/users": {"x-handler": "foo"}},
+        schema_name=schema_name,
+        paths=endpoint,
     )
     result = testdir.runpytest("-s")
     # Then it should be correctly processed

